Make fontification a little less 'constant-heavy'

theothornhill opened the pull request:
However, for this one, I'd like for you to take a look and test :)

This tries to lighten up a little on #169 with choosing variable-name-face, rather than the constant name face which is the default. This makes the fontification a little more similar to how vscode looks without directly disabling fontification.

Of course, these things are color scheme dependent, and also on taste and such. Personally, I agree on the constant face being a little heavy (on my colorscheme), and this lightens it up a little.

@josteink What do you think?

theothornhill commented:
However, for this one, I'd like for you to take a look and test :) 

This tries to lighten up a little on #169

theothornhill commented:
I also added some syntax highlighting for methods/functions. I think this serves ok as a starting point for syntax highlighting, without going too crazy all in.

josteink commented:
Not having yet considered the code, but just looking at how it behaves, I don't see anything wrong with this.

If all of the things now  fontitified with `font-lock-variable-face` should be or not is clearly something which can be discussed, but at the same time I don't think there's any clear, obvious or correct answer. Either way, it is clearly more correct than applying the `font-lock-constant-face` to them.

And with my theme at least, this looks much better.

josteink commented:
As for the code, I see you've updated some test-cases to match the new fontification applied.

When adding function-name fontification, wouldn't it be a good idea to add a test-case for that too?

theothornhill commented:
> As for the code, I see you've updated some test-cases to match the new fontification applied.
> 
> When adding function-name fontification, wouldn't it be a good idea to add a test-case for that too?

Yeah, I'll do that. I assume you agree with how that fontifies, at least for now then. I didn't add test cases yet since these matters definitively are up for suggestions/opinions

> Not having yet considered the code, but just looking at how it behaves, I don't see anything wrong with this.

The code should not be very controversial. It is the default value, without code specific to other languages, with the face changed. I hope I can as clean a solution as I did in the matchers-after variable, but had some issues with the byte compiled function in the defaults. Not sure what I need to do there yet.


> 
> If all of the things now fontitified with `font-lock-variable-face` should be or not is clearly something which can be discussed, but at the same time I don't think there's any clear, obvious or correct answer. Either way, it is clearly more correct than applying the `font-lock-constant-face` to them.

Yeah, I picked variable face since that usually isn't the most obnoxious color in most schemes, since it is used so often. 

> 
> And with my theme at least, this looks much better.

Good. Mine as well. I think this is a much better default as a starting point at least.

josteink commented:
Yeah. I'm happy with how function-highlighting looks and works, at least for the cases I've tested. 

I suspect it may not work with functions which accept generic type-arguments, but this is clearly an improvement and we should take one step at a time 🙂

theothornhill commented:
> Yeah. I'm happy with how function-highlighting looks and works, at least for the cases I've tested.

Great!

> 
> I suspect it may not work with functions which accept generic type-arguments, but this is clearly an improvement and we should take one step at a time slightly_smiling_face

I added the test for function fontification, also added naively optional regex to search for `<` or `(`, so that we catch both types of functions.

Are you satisfied with this for now with #169, or should we keep it open longer?

josteink commented:
We're in a phase of greenfield-ish experimentation and I think it makes sense to land clear improvements once they're "good enough". 

IMO that's the best way to determine what remaining parts needs improvements (if any) and if so which parts needs it the most. 

If you're happy, feel free to merge 🙂
